src/main/java/com/google/devtools/build/lib/analysis/starlark/StarlarkTransition.java

```diff
@@ -267,31 +267,38 @@ private static Target getActual(
       Map<PackageValue.Key, PackageValue> buildSettingPackages,
       Map<String, BuildOptions> toOptions)
       throws TransitionException {
-    // Collect settings changed during this transition and their types. This includes settings that
-    // were only used as inputs as to the transition and thus had their default values added to the
-    // fromOptions, which in case of a no-op transition directly end up in toOptions.
-    Map<Label, Rule> changedSettingToRule = Maps.newHashMap();
+    // Collect settings that are input or outputs of the transition together with their types.
+    // Output setting values will be validated and cleaned if set to the default.
```

Comment:
Can we quickly comment-clarify what "cleaned" means? i.e. it means "removed" vs. "normalized", "linted", etc.